pkgs/by-name/us/usque/package.nix

```
@@ -0,0 +1,60 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  stdenv,
+  nix-update-script,
+  buildPackages,
+  installShellFiles,
+  versionCheckHook,
+}:
+buildGoModule (finalAttrs: {
+  pname = "usque";
+  version = "1.4.1";
+  src = fetchFromGitHub {
+    owner = "Diniboy1123";
+    repo = "usque";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-/4/hsoWGdNe5vF22hFXuItXlYiLZ5BKrfg5XSFyljhE=";
+  };
+
+  vendorHash = "sha256-ngLlG9HP0KJPjIBVsNZuwnlJj2egMEh9U0xVQpVEg1Q=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/Diniboy1123/usque/cmd.version=${finalAttrs.version}"
+  ];
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) (
+    let
+      emulator = stdenv.hostPlatform.emulator buildPackages;
+    in
+    ''
+      installShellCompletion --cmd usque \
+        --bash <(${emulator} $out/bin/usque completion bash) \
+        --fish <(${emulator} $out/bin/usque completion fish) \
+        --zsh <(${emulator} $out/bin/usque completion zsh)
+    ''
+  );
```

generally emulators are frowned on unless absolutely required
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd usque \
      --bash <($out/bin/usque completion bash) \
      --fish <($out/bin/usque completion fish) \
      --zsh <($out/bin/usque completion zsh)
  '';
```